### PR TITLE
feat: Add relevant fields to the Explore Catalog CSV Download

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

- follow-on to #373
- be more defensive with accessing algolia results (missing keys)
- extend tests to cover edge-cases seen in PROD

## References

- [ENT-5295](https://openedx.atlassian.net/browse/ENT-5295)

### CODE REVIEW

File: enterprise_catalog/apps/api/v1/tests/test_views.py [hunk 1]
```diff
@@ -458,7 +458,21 @@ class EnterpriseCatalogCsvDataViewTests(APITestMixin):
             'upgrade_deadline': 32503680000.0,
         },
         'objectID': 'course-3543aa4e-3c64-4d9a-a343-5d5eda1dacf8-catalog-query-uuids-0'
-    }]}
+    },
+        {
+        'aggregation_key': 'course:MITx+19',
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/v1/tests/test_views.py:463]: test cases with missing `content_type` and `key` fields, which is something unexpected we saw in production algolia data

**Review by kiram15 — APPROVED** [review]
LGTM!

**Comment by binodpant** [comment]
looks like the commit message should have been a `fix: ...` instead? but yeah the change itself looks good to me. I wonder what those cases of missing key and/or content_type are... seems we are not good about enforcing schemas somewhere ( or maybe there are legit cases ?)

**Review by binodpant — APPROVED** [review]
looks good, just minor feedback/q

File: enterprise_catalog/apps/api/v1/tests/test_views.py (hunk 1 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api/v1/tests/test_views.py:463]: missing `content_type`s in production seems highly alarming. What were the cases that this happened?

File: enterprise_catalog/apps/api/v1/views.py [hunk 2]
```diff
@@ -380,25 +380,26 @@ def retrieve_indexed_data(self, facets, algoliaQuery):
             course_keys_chunk = []
             for hit in page.get('hits', []):
                 # ignore program data (for now)
-                if hit['content_type'] != 'course':
+                if hit.get('content_type') != 'course':
                     continue
-                course_keys_chunk.append(hit['key'])
+                if hit.get('key'):
```
Comment (alex-sheehan-edx) [inline enterprise_catalog/apps/api/v1/views.py:385]: won't this mess with ordering of the data if the key doesn't exist?

**Review by alex-sheehan-edx — APPROVED** [review]
two questions but otherwise looks good

File: enterprise_catalog/apps/api/v1/views.py (hunk 2 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api/v1/views.py:385]: oh I see! Nevermind

**Comment by johnnagro** [comment]
> missing content_types in production seems highly alarming. What were the cases that this happened?

agreed. since the button is "broken" given the current code and we're working on a wholesale replacement in parallel, I prioritized being defensive over chasing down exactly whats up in Algolia.

I will be looking a bit deeper for my parallel work.
